### ISSUE TITLE
Pipeline and pipelinerun lifecycle Update

### ISSUE THREAD

**Issue by chilianyi** [kubesphere/ks-devops#859]
### Discussed in https://github.com/kubesphere/ks-devops/discussions/856

<div type='discussions-op-text'>

<sup>Originally posted by **chilianyi** November 24, 2022</sup>
Pipeline Sequence Diagram

```mermaid
    sequenceDiagram
        actor A as User
        participant B as ApiServer
        participant C as K8s ApiServer
        participant D as Pipeline Controller
        participant E as Pipeline Metadata Controller
        participant F as Jenkins
        A ->> B : Create Pipeline
        B ->> C : Create Pipeline CR
        
        D -->> C : Get Pipeline CR
        activate D
        D ->> F : Create Jenkins Job
        deactivate D

        loop Every 10s
            E -->> E : Reconcile
        end
        E -->> F : Obtain Pipeline Metadata
        activate E
        E ->> C : Update Pipeline CR Metadata
        deactivate E

```

Pipeline Sync Status State Diagram

```mermaid
stateDiagram-v2
[*] --> Pending
Pending --> Successful
Successful --> [*]

```

PipelineRun Sequence Diagram

```mermaid
    sequenceDiagram
        actor A as User
        participant B as ApiServer
        participant C as K8s ApiServer
        participant D as PipelineRun Controller
        participant E as Jenkins
        A ->> B : Create PipelineRun
        B ->> C : Create PipelineRun CR
        
        D -->> C : Get PipelineRun CR
        activate D
        D ->> E : Trigger Jenkins Build
        D ->> C : Update PipelineRun CR ID
        deactivate D
        
        loop 
            D ->> E : Get PipelineRun Result
            activate D
            D ->> C : Update PipelineRun CR Status
            deactivate D
        end

```

PipelineRun Status State Diagram

```mermaid
stateDiagram-v2
[*] --> Pending
Pending --> Queue
Queue --> Running
Running --> Queue

Queue --> Aborted
Running --> Aborted

Running --> Succeeded
Running --> Failed
Aborted --> [*]
Succeeded --> [*]
Failed --> [*]

```

Problem1: When status is aborted, user can not view details
<img width="964" alt="截屏2022-11-24 13 07 32" src="https://user-images.githubusercontent.com/5917832/203698876-9e342125-99ba-48b2-95ba-fe32d704de21.png">

Problem2: When status is queue, pipelinerun diagram is not show
<img width="1440" alt="截屏2022-11-24 13 08 52" src="https://user-images.githubusercontent.com/5917832/203699057-5b93126f-5ff8-4393-bad8-5635b0994337.png">
While running will change to queue, so the diagram will show、disappear、show again...
<img width="1440" alt="截屏2022-11-24 13 09 46" src="https://user-images.githubusercontent.com/5917832/203699229-ed65df44-c07f-4bca-8bfd-bf78d6936cf9.png">

Maybe we could delete queue status.
```mermaid
stateDiagram-v2
[*] --> Pending
Pending --> Running
Running --> Aborted

Running --> Succeeded
Running --> Failed
Aborted --> [*]
Succeeded --> [*]
Failed --> [*]

```

</div>

**Comment by chilianyi** [comment]
<img width="1385" alt="截屏2022-12-01 10 23 27" src="https://user-images.githubusercontent.com/5917832/204950800-31815ba0-0c37-4ce2-ba39-426e6546f721.png">
<img width="1440" alt="截屏2022-12-01 10 23 16" src="https://user-images.githubusercontent.com/5917832/204950811-06ac0f5c-ea7a-4c32-ba8d-02b598cf4c7d.png">

When the pipelinerun status is aborted, it is not able to show url
While the url is able to open

Only pipelinerun in `not running` status is not able to open url

@harrisonliu5 please add link for `aborted` status pipelinerun

**Comment by wenxin-01** [comment]
Issue fixed in `kubespheredev/devops-apiserver:ks-v3.3.2-rc.0`
![image](https://user-images.githubusercontent.com/68640256/215378042-06b32259-0550-40e9-86ee-22bcea54f4ea.png)
